### PR TITLE
[DUOS-1692] Add reject ToS button

### PR DESCRIPTION
**Pull request by connorlbark**
## Addresses

https://broadworkbench.atlassian.net/browse/DUOS-1692

Adds 'Reject' button the ToS page, which allows a user to reject the terms after they've already logged in and accepted.

Requires backend on [this branch](https://github.com/DataBiosphere/consent/tree/DUOS-1691-reject-tos) as a new endpoint had to be created.

----
Have you read [Terra's Contributing Guide](https://github.com/DataBiosphere/terra-ui/wiki/Contributor-Guide) lately? If not, do that first.

- Label PR with a Jira ticket number and include a link to the ticket
- Label PR with a security risk modifier [no, low, medium, high]
- PR describes scope of changes
- Get a minimum of one thumbs worth of review, preferably two if enough team members are available
- Get PO sign-off for all non-trivial UI or workflow changes
- Verify all tests go green
- Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

File: src/libs/ajax.js [hunk 1]
```diff
@@ -1137,6 +1137,11 @@ export const ToS = {
     const url = `${await Config.getApiUrl()}/api/sam/register/self/tos`;
     const res = await axios.post(url, {}, Config.authOpts());
     return res.data;
+  },
+  rejectToS: async () => {
+    const url = `${await Config.getApiUrl()}api/sam/register/self/tos`;
```
Comment (rushtong) [inline src/libs/ajax.js]: Missing a slash:
```suggestion
    const url = `${await Config.getApiUrl()}/api/sam/register/self/tos`;
```

**Review by rushtong — CHANGES_REQUESTED** [review]
Looks and works perfectly for a signed in user. We need to hide this for users who have not signed in yet, so this needs that one tweak (and path fix in comments) before moving forward.

**Comment by connorlbark** [comment]
should be fixed now! slash added + button not rendered if logged out.

**Review by rushtong — APPROVED** [review]
Thank you for the changes 👍🏽

**Review by quazi-h — APPROVED** [review]
Changes look good to me and I was able to use the reject ToS button when signed in and I do not see it when signed out.

UI behavior is working as expected - when clicking the reject button, I am automatically signed out and when signing back in I am routed directly to the ToS page where I must accept to be able to access the UI. The reject button is grayed out if I have not made a decision yet and if I choose to reject here, I am signed out.
